rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactoryTest.java

```diff
@@ -108,6 +114,40 @@ public void testOnActivityResultShouldPopulateCorrectData() throws Exception {
         // verify correct action on back-press
         oneScanCovidRDTBarcodeFactory.onActivityResult(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE, Activity.RESULT_CANCELED, intent);
         Mockito.verify(formFragment).setMoveBackOneStep(true);
+
+        Intent invalidIntent = new Intent();
+        invalidIntent.putExtra(Constants.Config.ENABLE_BATCH_SCAN, true);
+        invalidIntent.putExtra("data", "");
+        oneScanCovidRDTBarcodeFactory.onActivityResult(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE, Activity.RESULT_OK, invalidIntent);
+        oneScanCovidRDTBarcodeFactory.onActivityResult(Activity.RESULT_OK, JsonFormConstants.BARCODE_CONSTANTS.BARCODE_REQUEST_CODE, null);
```

Comment:
Could we add verification for the expected behavior under a null and invalid intent?

Reply:
I have removed this code because the batch scan test verification is already done in this test case

Reply:
does this test for invalid and null intents?

Reply:
no